pkg/cmd/server/server_test.go

```diff
@@ -97,3 +98,39 @@ func TestArkResourcesExist(t *testing.T) {
 	arkAPIResourceList.APIResources = arkAPIResourceList.APIResources[:3]
 	assert.Error(t, server.arkResourcesExist())
 }
+
+func TestDefaultVolumeSnapshotLocations(t *testing.T) {
+	namespace := "heptio-ark"
+	arkClient := fakeclientset.NewSimpleClientset()
+	location := &v1.VolumeSnapshotLocation{
+		ObjectMeta: metav1.ObjectMeta{Name: "location1"},
+		Spec:       v1.VolumeSnapshotLocationSpec{Provider: "provider1"},
+	}
+	arkClient.ArkV1().VolumeSnapshotLocations(namespace).Create(location)
+
+	defaultVolumeSnapshotLocations := make(map[string]string)
+
+	// No defaults
+	volumeSnapshotLocations, err := getDefaultVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
+	assert.Equal(t, len(volumeSnapshotLocations), 0)
+	assert.NoError(t, err)
```

Comment:
shouldn't this case error per L334?

Reply:
No, line 317 (https://github.com/heptio/ark/pull/892/files#diff-cd1a009fa409db8496ea35b53f9776d2R317) makes this a NOOP. The flag parsing library itself enforces that an argument is actually passed in:

```An error occurred: flag needs an argument: --default-volume-snapshot-locations```

So for the sake of unit testing, this is the behavior I would expect.

Reply:
Ah, makes sense.